example/spec/discoverGranules/DiscoverGranulesHttpSpec.js

```diff
@@ -167,4 +175,118 @@ describe('The Discover Granules workflow with http Protocol', () => {
       });
     });
   });
+
+  describe('the DiscoverGranules Lambda', () => {
+    beforeAll(async () => {
+      await apiTestUtils.updateCollection({
+        prefix: config.stackName,
+        collection,
+        updateParams: { files: [] }
+      });
+
+      httpWorkflowExecution = await buildAndExecuteWorkflow(config.stackName,
+        config.bucket, workflowName, collection, provider);
+    });
+
+    it('encounters a collection without a files configuration', async () => {
+      const lambdaInput = await lambdaStep.getStepInput(
+        httpWorkflowExecution.executionArn, 'DiscoverGranules'
+      );
+
+      expect(lambdaInput.meta.collection.files).toEqual([]);
+    });
+
+    it('executes successfully', () => {
+      expect(httpWorkflowExecution.status).toEqual('SUCCEEDED');
+    });
+
+    it('discovers granules, but output has no files', async () => {
+      const lambdaOutput = await lambdaStep.getStepOutput(
+        httpWorkflowExecution.executionArn, 'DiscoverGranules'
+      );
+
+      expect(lambdaOutput.payload.granules.length).toEqual(3);
+      lambdaOutput.payload.granules.forEach((granule, i) => {
+        expect(granule.granuleId).toEqual(`granule-${i + 1}`);
+        expect(granule.files.length).toEqual(0);
+      });
+    });
+  });
+
+  describe('the DiscoverGranules Lambda', () => {
+    beforeAll(async () => {
+      await apiTestUtils.updateCollection({
+        prefix: config.stackName,
+        collection,
+        updateParams: { files: [collection.files[0]] }
+      });
+
+      httpWorkflowExecution = await buildAndExecuteWorkflow(config.stackName,
+        config.bucket, workflowName, collection, provider);
+    });
+
+    it('encounters a collection with a files configuration that does not match all files', async () => {
+      const lambdaInput = await lambdaStep.getStepInput(
+        httpWorkflowExecution.executionArn, 'DiscoverGranules'
+      );
+
+      expect(lambdaInput.meta.collection.files).toEqual([collection.files[0]]);
+    });
+
+    it('executes successfully', () => {
+      expect(httpWorkflowExecution.status).toEqual('SUCCEEDED');
+    });
+
+    it('discovers granules, but output does not include all files', async () => {
+      const lambdaOutput = await lambdaStep.getStepOutput(
+        httpWorkflowExecution.executionArn, 'DiscoverGranules'
+      );
+
+      expect(lambdaOutput.payload.granules.length).toEqual(3);
+      lambdaOutput.payload.granules.forEach((granule, i) => {
+        expect(granule.granuleId).toEqual(`granule-${i + 1}`);
+        expect(granule.files.length).toEqual(1);
+      });
+    });
+  });
+
+  describe('the DiscoverGranules Lambda', () => {
```

Comment:
I could do that, but I think it might get a bit awkward or require repetitive wording because the `it` calls that the `describe` call contains provide the distinct descriptions. Specifically:

```
  describe('the DiscoverGranules Lambda', () => {
    ...
    it('encounters a collection without a files configuration', async () => {
      ..

  describe('the DiscoverGranules Lambda', () => {
    ...
    it('encounters a collection with a files configuration that does not match all files', async () => {
      ...

  describe('the DiscoverGranules Lambda', () => {
    ...
    it('encounters a collection that has no files config, but should ignore files config', async () => {
      ...
```

However, if you have any suggestions for enhancing the `describe` descriptions without duplicating what's in the `it` descriptions, I'd be happy to update them.

Reply:
Different `describe` makes it easy to identify which test failures, especially when`beforeAll` fails